pyodide-build/pyodide_build/mkpkg.py

```diff
@@ -100,6 +105,48 @@ def _find_dist(
 
 def _get_metadata(package: str, version: str | None = None) -> MetadataDict:
     """Download metadata for a package from PyPI"""
+
+    if version is not None:
+        # first download metadata for current version so we
+        # can get the list of versions
+        url = f"https://pypi.org/pypi/{package}/json"
+        chosen_version = None
+        try:
+            with urllib.request.urlopen(url) as fd:
+                main_metadata = json.load(fd)
+                all_versions = main_metadata["releases"].keys()
+                all_versions = sorted(map(pkg_resources.parse_version, all_versions))
+                this_ver = pkg_resources.parse_version(version)
+                if this_ver in all_versions:
+                    chosen_version = str(this_ver)
+                else:
+                    try:
+                        spec = packaging.specifiers.SpecifierSet(version)
+                        chosen_version = None
+                        for v in reversed(all_versions):
+                            if spec.contains(str(v)):
```

Comment:
Maybe use `spec.filter(all_versions)` and then re-sort the results?

https://packaging.pypa.io/en/latest/specifiers.html#packaging.specifiers.SpecifierSet.filter 

It would be shorter.